kytos/core/user.py

```diff
@@ -25,17 +26,26 @@ def dict(self, **kwargs) -> dict:
         return values
 
 
+class HashDoc(BaseModel):
+    """HashDoc. Parameters for hash.scrypt function"""
+    salt: Optional[bytes] = os.urandom(16)
```

Comment:
@Alopalao, this will result in the same `salt` when this `BaseModel` is instantiated since `os.urandom(16)` will be evaluated at import time. Maybe you can use a `@validator('id', pre=True, always=True)` to set this dynamically if it's not set? `default_factory` would need a partial func and looks like it's not stable until pydantic version 2.

Reply:
Looks like when it is nested, `@validator('id', pre=True, always=True)` method does not work or at least I follow something similar to this [issue from pydantic](https://github.com/pydantic/pydantic/issues/866). I did it with `@root_validator` where first I populate `salt` and then `password` to ended with the `values` returned. Let me know if this is not ideal.

Reply:
Right. I followed up here https://github.com/kytos-ng/kytos/pull/312#discussion_r1043759844